### PR TITLE
[fix #1604] Commands send, request, location are sent several times if quickly tap on send button

### LINKED ISSUE #1604: Commands send, request, location are sent several times if quickly tap on send button
### Description

*Type*: Bug

*Summary*: Same message for commands (send, request, location) is sent several times if quickly tap on the send button in 1-1 or group chats. This does not happen for text or emoji. Expected: consistent behaviour for everything that is sent - anything in the input field is sent once even if user taps on send button several times.

#### Expected behavior
message for commands send, request, location is sent once if quickly tap on the send button.
#### Actual behavior
message for commands send, request, location is sent several times if quickly tap on the send button.

### Reproduction
video: https://drive.google.com/open?id=0Bz3t9zSg1wb7MGZhbGNJWkNoMGc
- Open Status
- Open 1-1 chat
- Construct a request to the contact in the input field, e.g. /request 0.1 
- quickly tap on send button several times
- check that messages area for new request messages (expected only 1)
- Construct send command in the input field, e.g. /send 0.3333
- quickly tap on send button several times
- Tap on /location command
- On the suggestions shown, quickly tap several times on the address

### Additional Information
* Status version: develop 0.9.9d130 / 0.9.8-130-g6a56b600 and 0.9.12
* Operating System: 
Real device Samsung Galaxy S6, Android 6.0.1
Real device iPhone 6s, iOS 10.3.3

### PR DESCRIPTION
**Pull request by alwx**
Fixes #1604 by doing the following:
1) adding a condition (there were no `:sending-in-progress?` check in a proper place)
2) fixing `src/status_im/chat/events/commands.cljs` (`:db` key was missing there)
  
Status: ready

### CODE REVIEW

**Comment by yenda** [comment]
Isn't it doing the same as https://github.com/status-im/status-react/pull/2663/files ?

**Comment by janherich** [comment]
@alwx As @yenda already pointed out, there is a PR fixing that problem already out for quite some time.

**Comment by alwx** [comment]
@yenda @janherich 
Are you sure it fixes exactly this issue? I don't see any changes to `proceed-command` in #2663, and these changes are required to make this fix work. 
#2663 obviously contains a `:sending-in-progress?` check in a right place, but it's not enough, and I think it's still possible to send multiple messages by tapping on a send button quickly there.

**Comment by yenda** [comment]
@alwx yes it was not enough good catch if it actually fixes it. Can you rebase your PR on the other one because it has refactoring that was asked in the review

**Comment by alwx** [comment]
sure, will do it 
(I hope in can wait for about an hour because your one is not merged yet anyway).

File: src/status_im/chat/events/commands.cljs [hunk 1]
```diff
@@ -39,7 +39,8 @@
             to          (get-in contacts [chat-id :address])
             jail-params {:parameters params
                          :context    (generate-context current-account-id chat-id to group-id)}]
-        {:call-jail {:jail-id                 jail-id
+        {:db        db
```
Comment (janherich) [inline src/status_im/chat/events/commands.cljs:42]: What's the reason for including `:db` effect here ?

**Review by janherich — CHANGES_REQUESTED** [review]

**Comment by yenda** [comment]
@alwx yes it can wait, I think we should notifiy QA that they should test this one only since it will be based on the other one and actually fixes the problem

File: src/status_im/chat/events/commands.cljs (hunk 1 above)
Comment (alwx) [reply src/status_im/chat/events/commands.cljs:42]: because we modify it at an earlier stage and we need these modifications to be there

Comment (janherich) [reply src/status_im/chat/events/commands.cljs:42]: Where is that ? I fail to see it in the function or in any place where function is called.

Comment (alwx) [reply src/status_im/chat/events/commands.cljs:42]: `status-im.chat.events.input/command-complete-fx` -> `status-im.chat.events.input/proceed-command` -> `status-im.chat.events.input/request-command-data` -> `status-im.chat.events.commands/request-command-message-data`

**Comment by alwx** [comment]
@yenda done. it's a small 3-line commit now

**Review by janherich — APPROVED** [review]

**Comment by annadanchenko** [comment]
After sending request command in 1-1 chat keyboards hides. keyboard should stay open, so use can proceed typing after sending the command (as it is in develop build now)
both iOS and Android

Logs/Video (from 00:31): https://app.testfairy.com/projects/4803622-status/builds/7470181/sessions/2/?accessToken=PS5GcqApPIC/cNGWnISK8VApWdU

on Branch: PR-2988
Android: https://i.diawi.com/innYGq
iOS: https://i.diawi.com/SRXChk

**Comment by alwx** [comment]
@annadanchenko fixed

**Comment by oskarth** [comment]
Merged to develop. Cherry picked into release branch.

**Pull request merged**